From 1321f34f1ebcff61ad7e65e507cfd3e9028af19b Mon Sep 17 00:00:00 2001
From: vijay kumar <vpendo@codeaurora.org>
Date: Fri, 27 Mar 2015 12:13:42 +0530
Subject: app: aboot: add the integer overflow checks on sparse header

Change for checking interger overflow on sparse header partition
size calculation.

Change-Id: Ib0b613f2632645ef8edd8fa28c8f8ac21f3b982c
---
 app/aboot/aboot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/aboot/aboot.c b/app/aboot/aboot.c
index 1498e98..c90209d 100644
--- a/app/aboot/aboot.c
+++ b/app/aboot/aboot.c
@@ -2092,7 +2092,7 @@ void cmd_flash_mmc_sparse_img(const char *arg, void *data, unsigned sz)
 
 	/* Read and skip over sparse image header */
 	sparse_header = (sparse_header_t *) data;
-	if ((sparse_header->total_blks * sparse_header->blk_sz) > size) {
+	if (((uint64_t)sparse_header->total_blks * (uint64_t)sparse_header->blk_sz) > size) {
 		fastboot_fail("size too large");
 		return;
 	}
-- 
cgit v1.1

